### ISSUE TITLE
TypeError: must be type, not None

### ISSUE THREAD

**Issue by sebastianneubauer** [blue-yonder/pyscaffold#136]
On debian since newer versions of pyscaffold are available on pypi I get this error if pyscaffold is not installed in the virtualenv (pyscaffold gets installed via easy install as egg (:rolling_eyes:) ) before installing the project in dev mode `pip install -e .`:

```
pip install -e .
Obtaining file:///vagrant_data/
    Complete output from command python setup.py egg_info:

    Installed /vagrant_data/.eggs/PyScaffold-2.5.8-py2.7.egg
    [pbr] Generating ChangeLog
    ERROR:root:Error parsing
    Traceback (most recent call last):
      File "/vagrant_data/.eggs/PyScaffold-2.5.8-py2.7.egg/pyscaffold/contrib/pbr/pbr/core.py", line 111, in pbr
        attrs = util.cfg_to_args(path, dist.script_args)
      File "/vagrant_data/.eggs/PyScaffold-2.5.8-py2.7.egg/pyscaffold/contrib/pbr/pbr/util.py", line 267, in cfg_to_args
        wrap_commands(kwargs)
      File "/vagrant_data/.eggs/PyScaffold-2.5.8-py2.7.egg/pyscaffold/contrib/pbr/pbr/util.py", line 569, in wrap_commands
        cmdclass = ep.resolve()
      File "/home/vagrant/venv/local/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2297, in resolve
        module = __import__(self.module_name, fromlist=['__name__'], level=0)
      File "/home/vagrant/venv/local/lib/python2.7/site-packages/sphinx/setup_command.py", line 23, in <module>
        from sphinx.application import Sphinx
      File "/home/vagrant/venv/local/lib/python2.7/site-packages/sphinx/application.py", line 33, in <module>
        from sphinx.environment import BuildEnvironment
      File "/home/vagrant/venv/local/lib/python2.7/site-packages/sphinx/environment/__init__.py", line 46, in <module>
        from sphinx.util.websupport import is_commentable
      File "/home/vagrant/venv/local/lib/python2.7/site-packages/sphinx/util/websupport.py", line 11, in <module>
        from sphinxcontrib.websupport.utils import is_commentable  # NOQA
      File "/home/vagrant/venv/local/lib/python2.7/site-packages/sphinxcontrib/websupport/__init__.py", line 15, in <module>
        from sphinxcontrib.websupport.core import WebSupport  # NOQA
      File "/home/vagrant/venv/local/lib/python2.7/site-packages/sphinxcontrib/websupport/core.py", line 17, in <module>
        from jinja2 import Environment, FileSystemLoader
      File "/home/vagrant/venv/local/lib/python2.7/site-packages/jinja2/__init__.py", line 33, in <module>
        from jinja2.environment import Environment, Template
      File "/home/vagrant/venv/local/lib/python2.7/site-packages/jinja2/environment.py", line 15, in <module>
        from jinja2 import nodes
      File "/home/vagrant/venv/local/lib/python2.7/site-packages/jinja2/nodes.py", line 19, in <module>
        from jinja2.utils import Markup
      File "/home/vagrant/venv/local/lib/python2.7/site-packages/jinja2/utils.py", line 486, in <module>
        MutableMapping.register(LRUCache)
      File "/home/vagrant/venv/lib/python2.7/abc.py", line 109, in register
        if issubclass(subclass, cls):
      File "/home/vagrant/venv/lib/python2.7/abc.py", line 180, in __subclasscheck__
        if issubclass(subclass, scls):
      File "/home/vagrant/venv/local/lib/python2.7/site-packages/typing.py", line 1265, in __subclasscheck__
        return super(GenericMeta, self).__subclasscheck__(cls)
    TypeError: must be type, not None
    error in setup command: Error parsing /vagrant_data/setup.cfg: TypeError: must be type, not None
```

Installing pyscaffold in the virtualenv before `pip install -e. ` solves the problem but is somehow ugly and doesn't really fit in my workflow...

Anything I can do to circumvent this or is this a bug?

**Comment by FlorianWilhelm** [comment]
Hi, we had this problem once on a Travis build and solved it by pinning Sphinx to version 1.5.6. We could only find out that it has something to do with Sphinx and most likely PBR which is used by PyScaffold 2.5.* in the way it monkey patches into setuptools. 
In the current beta of PyScaffold 3 we got rid of PBR and also of the problem you describe. PBR does a lot of black magic and really is a nightmare, that's why we kicked it. 
That being said, I could never reproduce the problem locally on my machine. There are two things you could do to help us out extremely:

1) Describe how we can reproduce the problem by using PyScaffold 2.5.*
2) Upgrade your package to the current beta of PyScaffold 3 and tell us if you still go the problem. 

Your feedback is greatly appreciated!

**Comment by sebastianneubauer** [comment]
It can be worked around by installing pyscaffold using pip before `pip install -e.`:

```
pip install pyscaffold==2.5.7
pip install -e.
```
but this is really just a workaround...we should fix this somehow...but I have no clue...

**Comment by FlorianWilhelm** [comment]
@sebastianneubauer Thanks, this is a workaround but what about the other points from my comment above? Does this also happen if you fix Sphinx to version 1.5.6? Can you reproduce it also directly on your laptop with a minimal example so that I can reproduce it too? I would really like to understand more but so far I could not reproduce it. I agree that a workaround like this should not be necessary.

**Comment by FlorianWilhelm** [comment]
This error could not be reproduced on the current master which will be released as version 3.0 soon. Closing this issue for now.
